tests/test_main.py

```diff
@@ -1247,7 +1250,7 @@ def check_code(self, code, nodes, decorators, check_names):
             # The relation between `ast.Name` and `argval` is already
             # covered by the verifier and much more complex in python 3.11 
```

Comment:
A bit off topic, but mangling of variable names has always been there and I just didn't realise it, right? It didn't become more complex in 3.11?

Reply:
I learned a lot of new stuff during this pull-request, but I don't know how it has been before. Name mangeling also surprised me

Reply:
I experimented a bit:

I can not think of any use for mangled local variables in real code.

So I tried if it is possible to use mangled arguments and the result surprised me.

script:
```python
class A:
    def a(self, *, __private): # argument is mangled
        print(__private)

    def b(self, v):
        self.a(__private=v) # call is not mangled

    def c(self, v):
        self.a(_A__private=v) # we have to call it explicit


a = A()

a.c("c")
a.a(_A__private="a")
a.b("b")

```

output (Python 3.11.0rc2+):
```python
c
a
Traceback (most recent call last):
  File "/home/frank/projects/executing/except_test.py", line 16, in <module>
    a.b("b")
  File "/home/frank/projects/executing/except_test.py", line 6, in b
    self.a(__private=v) # call is not mangled
    ^^^^^^^^^^^^^^^^^^^
TypeError: A.a() got an unexpected keyword argument '__private'
```
I don't think that this is intentional, because it is not possible to call this function with the `__private` argument.

Reply:
This is an excellent point. It already doesn't feel intentional just because it's weird, but this is a solid example of a bad outcome, `self.a(__private=v)` not working can only be fairly described as a bug and should be reported as such. On the other hand, it seems very low priority, because no one does this in practice, to the extent that we didn't know about it. I wouldn't be surprised if there's already a report about this and no one really cares. But I might be wrong.

Reply:
found an interresting real world example:
``` python
class Opt:
    __optionalNotMatched = _NullToken()

    def __init__(
        self, expr: Union[ParserElement, str], default: Any = __optionalNotMatched
    ):
        ...
```
there are also several other

``` bash
$ rg " __[A-Za-z_]*[^_,:] " **.py
```

I don't think that it is worth thinking to much about that stuff.

ok ... a last one :-) 

``` python
# name mangling
class TestMangling:

    def test(self):
        try:
            raise Exception(10)
        except Exception as __e:
            assert __e == __e

```